### PR TITLE
🎉 Source TikTok Marketing: add optional `atttribution window` 

### LINKED ISSUE #23756: Source TikTok Marketing: fix `conversion` metric values by adding a lookback window
## Environment
- **Source Connector and version**: TikTok Marketing `2.0.3` (latest as of today)
- **Step where error happened**: After sync job (which is successful), when data modelling for report

## Current Behavior
For any `*_report_daily` stream, all metrics are legit except `conversion` since this metric may change after few days. 

## Expected Behavior
It's typical for a marketing API to have what we call an attribution window, which is a number of days while the metric can change after an action (especially `conversion` as it is something that can happen way after seeing an ad).

One example of an Airbyte connector with this logic is Facebook Marketing (in [README](https://github.com/airbytehq/airbyte/tree/master/airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing#insights), in [spec](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/spec.py#L80), in [stream definition](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/streams/base_insight_streams.py#L242))
We should have a default value for attribution window (maybe `7` for TikTok as it looks like it's the default one?), that can be changed at the source setup as the app allows custom attribution windows.

## Additional information
To achieve this, we usually need:
- To have `Incremental | Deduped + History` with the right composite key (which may not be the case, cf issue [here](https://github.com/airbytehq/airbyte/issues/23728) -> not fixing this before may lead to unexpected results implementing the `lookback_window`)
- To have a `lookback_window` that under the hood does something like `start_date_from_state - timedelta(days=lookback_window_in_days)`

## Steps to Reproduce
1. Run a successful job in `Incremental | Deduped + History` mode for any `_report_daily` stream (more specifically you can try `ad_groups_reports_daily`). `conversion` metric looks aligned with reports in the TikTok reports UI.
2. Wait for a day or two, rerun the job and look at past data. Now `conversion` metric is lower in your table than in the TikTok Business UI, because some people converted with your ad sync your first sync but the data has never been refreshed.

## Are you willing to submit a PR?
Not a the moment.

### PR DESCRIPTION
**Pull request by bazarnov**
## What
Resolving: https://github.com/airbytehq/airbyte/issues/23756

## How
- added `attribution_window` to the `spec`
- added `attribution_window` option passed to the `BasicReports` streams

## 🚨 User Impact 🚨
No impact is expected,  since the default value for `attr_window` is 0,  so no adjusting is applied by default.

## Pre-merge Checklist
*Expand the relevant checklist and delete the others.*

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Connector version has been incremented
    - [ ] Version has been bumped according to our [Semantic Versioning for Connectors](https://docs.airbyte.com/contributing-to-airbyte/semantic-versioning-for-connectors) guidelines
    - [ ] `Dockerfile` has updated version
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` with an entry for the new version. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)

- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

### CODE REVIEW

**Review by lazebnyi — COMMENTED** [review]
LGTM! I will approve after PR with migration will be merged.

**Review by lazebnyi — APPROVED** [review]

**Review by grubberr — APPROVED** [review]

File: airbyte-integrations/connectors/source-tiktok-marketing/source_tiktok_marketing/spec.json
Comment (grubberr) [inline airbyte-integrations/connectors/source-tiktok-marketing/source_tiktok_marketing/spec.json:98]: Can be order 3 ?

Comment (bazarnov) [reply airbyte-integrations/connectors/source-tiktok-marketing/source_tiktok_marketing/spec.json:98]: Since the `report_granularity` is gone now, I'll adjust this.

**Pull request merged**
